bundles/org.eclipse.osgi/container/src/org/eclipse/osgi/framework/util/Wirings.java

```diff
@@ -0,0 +1,120 @@
+/*******************************************************************************
+ *  Copyright (c) 2022, 2022 Hannes Wellmann and others.
+ *
+ *  This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License 2.0
+ *  which accompanies this distribution, and is available at
+ *  https://www.eclipse.org/legal/epl-2.0/
+ *
+ *  SPDX-License-Identifier: EPL-2.0
+ * 
+ *  Contributors:
+ *     Hannes Wellmann - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.osgi.framework.util;
+
+import static java.util.function.Predicate.not;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.eclipse.osgi.container.ModuleContainer;
+import org.eclipse.osgi.internal.framework.SystemBundleActivator;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.namespace.HostNamespace;
+import org.osgi.framework.namespace.IdentityNamespace;
+import org.osgi.framework.wiring.BundleRevision;
+import org.osgi.framework.wiring.BundleWire;
+import org.osgi.framework.wiring.BundleWiring;
+import org.osgi.framework.wiring.FrameworkWiring;
+import org.osgi.resource.Namespace;
+import org.osgi.resource.Requirement;
+
+public class Wirings { // Wiring Util
+
+	private Wirings() {
+	}
+
+	// TODO: check usage
+
+	public static Stream<Bundle> getBundles(String symbolicName) {
```

Comment:
```suggestion
	public static Stream<Bundle> getBundles(BundleContext context, String symbolicName) {
```
I would suggest to simply force users passing a BundleContext, given that one has to fetch the `PackageAdmin` before anyways there must be a bundlecontext somewhere around, and the BC can be queried for the systembundle like BundleContext.getBundle(Constants.SYSTEM_BUNDLE) ...